internal/rest/client/cluster.go

```diff
@@ -62,10 +62,10 @@ func (c *Client) ResetClusterMember(ctx context.Context, name string, force bool
 }
 
 // UpdateClusterCertificate sets a new cluster keypair and CA.
-func (c *Client) UpdateClusterCertificate(ctx context.Context, args apiTypes.ClusterCertificatePut) error {
+func (c *Client) UpdateClusterCertificate(ctx context.Context, name string, args apiTypes.KeyPair) error {
```

Comment:
I'm thinking from an external project's use case, they probably won't know to call `UpdateClusterCertificate` with `"cluster"` as an argument if they want to change their core certificate. 

So either we should document this in the client functions, and move `types.ClusterCertificate` into the public types package, or we should add a second client function `UpdateCoreClusterCertificate()` which doesn't take the name and internally uses `types.ClusterCertificate`.

Reply:
I am a fan of the latter as having another client func doesn't hurt and it's presumably easier to follow than reading the doc string of the function. Will add another one.

Reply:
After thinking a bit more about it I went with using a specific type as it feels more intuitive:

```go
func (m *MicroCluster) UpdateCertificate(ctx context.Context, name types.CertificateName, keypair types.KeyPair) error {}

func (c *Client) UpdateCertificate(ctx context.Context, name apiTypes.CertificateName, args apiTypes.KeyPair) error {}
```

I have also removed `cluster` from the functions name as this might be confusing now.
If you have custom cert you can now derive a new type from `type CertificateName string` and supply it to the function.
Or use `const ClusterCertificateName CertificateName = "cluster"` in case you want to update the cluster one.

Reply:
Fixed